scipy/stats/tests/test_bootstrap.py

```diff
@@ -54,6 +58,68 @@ def statistic(x, y, axis):
     with pytest.raises(ValueError, match=message):
         bootstrap(([1, 2, 3],), np.mean, random_state='herring')
 
+@pytest.mark.parametrize("method", ['basic', 'percentile', 'BCa'])
+def test_bootstrap_paired(method):
+    # test that `paired` works as expected
+    np.random.seed(0)
+    n = 100
+    x = np.random.rand(n)
+    y = np.random.rand(n)
+
+    def my_statistic(x, y, axis=-1):
+        return ((x-y)**2).mean(axis=axis)
+
+    def my_paired_statistic(i, axis=-1):
+        a = x[i]
+        b = y[i]
+        res = my_statistic(a, b)
+        return res
+    i = np.arange(len(x))
+
+    res1 = bootstrap((i,), my_paired_statistic, random_state=0)
+    res2 = bootstrap((x, y), my_statistic, paired=True, random_state=0)
+
+    assert_allclose(res1.confidence_interval, res2.confidence_interval)
+    assert_allclose(res1.standard_error, res2.standard_error)
+
+@pytest.mark.parametrize("method", ['basic', 'percentile', 'BCa'])
+@pytest.mark.parametrize("axis", [0, 1, 2])
+@pytest.mark.parametrize("paired", [True, False])
+def test_bootstrap_vectorized(method, axis, paired):
+    # test that paired is vectorized as expected: when samples are tiled,
+    # CI and standard_error of each axis-slice is the same as those of the
+    # original 1d sample
+
+    if not paired and method=='BCa':
+        # should re-assess when BCa is extended
+        pytest.xfail(reason="BCa currently for 1-sample statistics only")
+    np.random.seed(0)
+
+    def my_statistic(x, y, z, axis=-1):
+        return x.mean(axis=axis) + y.mean(axis=axis) + z.mean(axis=axis)
+
+    shape = 10, 11, 12
+    n_samples = shape[axis]
+
+    x = np.random.rand(n_samples)
+    y = np.random.rand(n_samples)
+    z = np.random.rand(n_samples)
+    res1 = bootstrap((x, y, z), my_statistic, paired=paired, method=method,
+                     random_state=0, axis=0, n_resamples=100)
+
+    reshape = [1, 1, 1]
+    reshape[axis] = n_samples
+    x = np.broadcast_to(x.reshape(reshape), shape)
+    y = np.broadcast_to(y.reshape(reshape), shape)
+    z = np.broadcast_to(z.reshape(reshape), shape)
+    res2 = bootstrap((x, y, z), my_statistic, paired=paired, method=method,
+                     random_state=0, axis=axis, n_resamples=100)
+
+    assert_allclose(res2.confidence_interval.low,
+                    res1.confidence_interval.low)
+    assert_allclose(res2.confidence_interval.high,
+                    res1.confidence_interval.high)
+    assert_allclose(res2.standard_error, res1.standard_error)
```

Comment:
Two blank lines before next `def`
```suggestion
    assert_allclose(res2.standard_error, res1.standard_error)

```